ci/scripts/cpp_test.sh

```diff
@@ -92,7 +92,9 @@ fi
 if [ "${ARROW_USE_MESON:-OFF}" = "ON" ]; then
   ARROW_BUILD_EXAMPLES=OFF # TODO: Remove this
   meson test \
+    --no-rebuild \
```

Comment:
Is the no-rebuild still required?

Reply:
Yes. If we don't have this, `meson test` re-downloads `cpp/subprojects/`. It produces unreadable files under source directory. It causes an `actions/cache` error.